pkg/controller/installation/core_controller.go

```diff
@@ -453,6 +453,25 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
+	log.Info("Creating secret in core_controller")
+
+	if instance.Spec.Variant == operator.TigeraSecureEnterprise && instance.Spec.ClusterManagementType == operator.ClusterManagementTypeManagement {
+		// Generate CA and TLS certificate for tigera-manager
+		_, err = render.CreateOperatorTLSSecret(nil,
+			render.ManagerTLSSecretName,
+			render.ManagerSecretKeyName,
+			render.ManagerSecretCertName,
+			825*24*time.Hour, // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
+			nil,
+			render.ManagerServiceExternalDNS,
+			render.ManagerServiceInternalDNS,
+		)
+		if err != nil {
+			r.SetDegraded("Error generating self signed certificates for tigera-manager", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
Looks like we reproduce the tigera-manager cert secret for Compliance and for Kube controller, using two different ways. 

Compliance we use `utils.ValidateCertPair()` and then pass to Compliance render, where it does an actual copy of the secret. 

Kube controllers, we do this `render.CreateOperatorTLSSecret()` directly. 

Was there a specific reason for the difference? I was going to guess that it's because Compliance component has its own render (but then it looks like Kube controller does as well).

Reply:
Compliance does not create the secret, it just copies it in its namespace. The core-controllers creates the secret and that make it available to be other components.

Reply:
My question is then: 
- Why does core-controllers need to create the secret and make it available? 

Before this Manager needed the secret, which it must have gotten a copy from getting it passed from the tigera operator namespace. So why does core-controller need to create its own (instead of a copy). 

I understand that Compliance and Kube controller both now need access to this Manager TLS secret, but why does core controller need to create a new version? 

This is more for my understanding, since I'm missing a step here.